BTTVEmoteProvider.cs

```diff
@@ -54,8 +54,10 @@ public async Task Initialize(string Channel)
 			void addToList(string Name, Stream EmoteStream)
 			{
 				var img = Image.Load(EmoteStream);
-				lock (EmoteCache) {
-					EmoteCache.Add(Name, new SizeF(img.Width, img.Height));
+				if (!EmoteCache.ContainsKey(Name)) {
+					lock (EmoteCache) {
```

Comment:
Please move `lock` to encapsulate `ContainsKey` call too, as emote could be added before the `lock`, but after `ContainsKey` resulting in duplicate entry.